Add Grid-in-Window KB article (#390)
telerik/blazor-docs: knowledge-base/window-expand-on-drag.md

---
title: Window Expands When Dragged
description: The Blazor TelerikWindow component grows the first time it is dragged. Its right border sticks to the right side of the screen.
type: troubleshooting
page_title: Window Expands When Dragged for the First Time
slug: window-kb-expand-on-drag
position: 
tags: window, expand, drag, grid, move
ticketid: 1522796, 1524154
res_type: kb
---

## Description

I have a Window with no fixed width and the `Draggable` property is set to true. Initially the Window is rendered to be as wide as its content.
When the user starts dragging the component for the first time, it grows and sticks to the right side of the screen.

## Cause\Possible Cause(s)

The Window may expand on drag if it has no fixed width, and its content has no width either.

> The Window is an absolutely positioned element. If it has no width, it will expand to enclose its content. If the Window content can expand horizontally without a limit, this can cause the Window to expand until it reaches the viewport boundary.

Author: ```suggestion
The Window is an absolutely positioned element. If it has no width, it will expand to enclose its content. If the Window content can expand horizontally without a limit (like block-level elements do), this can cause the Window to expand until it reaches the viewport boundary.
```

The unexpected expansion does not occur while the Window is centered, because of the styles, which the component uses at that time: `top: 50%`, `left: 50%` and especially `transform: translate(-50%, -50%);`. As soon as the user starts dragging the Window, the `transform` style is removed, which triggers different browser behavior.

### Grid inside a Window

> A specific scenario, which can cause the discussed behavior, is a Grid placed inside a Window.

Author: ```suggestion
The described behavior will occur if the Window contains a Grid.
```
